src/schemas/server.ts

```diff
@@ -0,0 +1,154 @@
+
+    /**
+     * The instance data from the database.
+     */
+    declare private data: Document;
+
+    /**
+     * Creates a new document for a server.
+     * 
+     * @param id server ID for the new document
+     */
+    private constructor(id: string);
+
+    /**
+     * Inflates a server object using data from the database.
+     * 
+     * @param data server data from database
+     */
+    private constructor(data: Document);
+
+    /**
+     * Instantiates a server object that represents a server document.
+     * 
+     * @param arg the data passed
+     */
+    private constructor(arg: string | Document) {
+        if (typeof arg === "string" || arg instanceof String) {
+            // Create new server document if a string was passed
+            this.data = new Server.model({
+                _id: <string> arg,
+                logs: null
+            });
+        } else {
+            // Otherwise use the document from the database
+            this.data = <Document> arg;
+        }
+    }
+
+    /**
+     * Gets a server document from the database.
+     * 
+     * @param id the server ID
+     * @returns the requested server document or null if none was found
+     */
+    public static async get(id: string): Promise<Server> {
+        // Try to get the server from the cache
+        const s: [Server, NodeJS.Timeout] = Server.cache.get(id);
+        let server: Server;
+
+        if (s) {
+            // If the server is found, clear the old timeout
+            clearTimeout(s[1]);
+            server = s[0];
+        } else {
+            // Otherwise, fetch from the database
+            const data: Document = await Server.model.findById(id);
+            
+            // If the database didn't have the server,
+            // create a new document
+            server = data ? new Server(data) : new Server(id);
+        }
+
+        // Create a timeout that flushes the server from the cache
+        // after a set period of time
+        const timeout: NodeJS.Timeout = setTimeout(async (): Promise<void> => {
+            await server.save();
+            Server.cache.delete(id);
+        }, serverLifetime);
+
+        // Save to the cache
+        Server.cache.set(id, [server, timeout]);
+
+        return server;
+    }
+
+    /**
+     * Saves the cached data to the database.
+     */
+    public async save(): Promise<void> {
+        await this.data.save();
+    }
+
+    /**
+     * Deletes the server configurations from the database.
+     */
+    public async delete(): Promise<void> {
+        // Clear the server data from the cache
+        clearTimeout(Server.cache.get(this.data._id)[1]);
+        // Delete from the database
+        await Server.model.findByIdAndDelete(this.data._id);
+    }
+
+    /**
+     * The ID of the logs channel
+     */
+    public get logs(): string {
+        return (<ServerT> this.data).logs;
+    }
+
+    public set logs(id: string) {
+        (<ServerT> this.data).logs = id;
+    }
+}
```

Comment:
This is the wrapper class for the database schema, it has all the necessary functions and methods